src/platform/utilities/ui/focus.js

```diff
@@ -0,0 +1,97 @@
+// .nav-header > h2 contains "Step {index} of {total}: {page title}"
+export const defaultFocusSelector = '.nav-header > h2';
+
+/**
+ * Focus on element
+ * @param {String|Element} selectorOrElement - CSS selector or attached DOM
+ *  element
+ * @param {FocusOptions} options - "preventScroll" or "focusVisible". See
+ *  https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/focus#parameters
+ * @param {Element} root - root element for querySelector; would allow focusing
+ *  on elements inside of shadow dom
+ */
+export function focusElement(selectorOrElement, options, root) {
+  const el =
+    typeof selectorOrElement === 'string'
+      ? (root || document).querySelector(selectorOrElement)
+      : selectorOrElement;
+  if (el) {
+    // Use getAttribute to grab the "tabindex" attribute (returns string), not
+    // the "tabIndex" property (returns number). Focusable elements will
+    // automatically have a tabIndex of zero, otherwise it's -1.
+    const tabindex = el.getAttribute('tabindex');
+    // No need to add, or remove a tabindex="0"
+    if (el.tabIndex !== 0) {
+      el.setAttribute('tabindex', '-1');
+      if (typeof tabindex === 'undefined' || tabindex === null) {
+        // Remove tabindex on blur. If a web-component is focused using a -1
+        // tabindex and is not removed on blur, the shadow elements inside will
+        // not be focusable
+        el.addEventListener(
+          'blur',
+          () => {
+            el.removeAttribute('tabindex');
+          },
+          { once: true },
+        );
+      }
+    }
+    el.focus(options);
+  }
+}
+
+/**
+ * Web components may not have their shadow DOM rendered right away, so we need
+ * to wait & check before setting focus on the selector; if not found after max
+ * iterations, then fall back to the default selector (step _ of _ h2)
+ * Discussion: https://dsva.slack.com/archives/CBU0KDSB1/p1676479946812439
+ * @param {String} selector - focus target selector
+ * @param {Element} root - starting element of the querySelector; may be a
+ *  shadowRoot
+ * @example waitForRenderThenFocus('h3', document.querySelector('va-radio').shadowRoot);
+ */
+export function waitForRenderThenFocus(
+  selector,
+  root = document,
+  timeInterval = 250,
+) {
+  const maxIterations = 6; // 1.5 seconds
+  let count = 0;
```

Comment:
Did you do any throttled-down speed testing with this? Just want to ensure at slower web speeds this doesn't inadvertently timeout.

Reply:
Another good call, I will do that!

Reply:
After the app loaded, I set network speed to "slow 3G" and "offline" and neither effected the web component rendering. I think it's more about processor speed. I see this delay if I have webpack actively running and I reload the page I'm on.